internal/alb/sg/instance_attachment.go

```diff
@@ -0,0 +1,193 @@
+)
+
+// InstanceAttachment represents the attachment of securityGroups to instance
+type InstanceAttachment struct {
+	GroupID string
+	Targets tg.TargetGroups
+}
+
+// InstanceAttachementController manages InstanceAttachment
+type InstanceAttachementController interface {
+	Reconcile(*InstanceAttachment) error
+	Delete(*InstanceAttachment) error
+}
+
+type instanceAttachmentController struct {
+	store store.Storer
+	ec2   *albec2.EC2
+}
+
+func (controller *instanceAttachmentController) Reconcile(attachment *InstanceAttachment) error {
+	instanceENIs, err := controller.getClusterInstanceENIs()
+	if err != nil {
+		return fmt.Errorf("failed to get cluster enis due to %s", err.Error())
+	}
+	supportingENIs := controller.findENIsSupportingTargets(instanceENIs, attachment.Targets)
+	for _, enis := range instanceENIs {
+		for _, eni := range enis {
+			if _, ok := supportingENIs[aws.StringValue(eni.NetworkInterfaceId)]; ok {
+				err := controller.ensureSGAttachedToENI(attachment.GroupID, eni)
+				if err != nil {
+					return err
+				}
+			} else {
+				err := controller.ensureSGDetachedFromENI(attachment.GroupID, eni)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func (controller *instanceAttachmentController) Delete(attachment *InstanceAttachment) error {
+	instanceENIs, err := controller.getClusterInstanceENIs()
+	if err != nil {
+		return fmt.Errorf("failed to get cluster enis, Error:%s", err.Error())
+	}
+	for _, enis := range instanceENIs {
+		for _, eni := range enis {
+			err := controller.ensureSGDetachedFromENI(attachment.GroupID, eni)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func (controller *instanceAttachmentController) ensureSGAttachedToENI(sgID string, eni *ec2.InstanceNetworkInterface) error {
+	desiredGroups := []string{sgID}
+	for _, group := range eni.Groups {
+		groupID := aws.StringValue(group.GroupId)
+		if groupID == sgID {
+			return nil
+		}
+		desiredGroups = append(desiredGroups, groupID)
+	}
+	_, err := controller.ec2.ModifyNetworkInterfaceAttribute(&ec2.ModifyNetworkInterfaceAttributeInput{
+		NetworkInterfaceId: eni.NetworkInterfaceId,
+		Groups:             aws.StringSlice(desiredGroups),
+	})
+	return err
+}
+
+func (controller *instanceAttachmentController) ensureSGDetachedFromENI(sgID string, eni *ec2.InstanceNetworkInterface) error {
+	sgAttached := false
+	desiredGroups := []string{}
+	for _, group := range eni.Groups {
+		groupID := aws.StringValue(group.GroupId)
+		if groupID == sgID {
+			sgAttached = true
+		} else {
+			desiredGroups = append(desiredGroups, groupID)
+		}
+	}
+	if !sgAttached {
+		return nil
+	}
+	_, err := controller.ec2.ModifyNetworkInterfaceAttribute(&ec2.ModifyNetworkInterfaceAttributeInput{
+		NetworkInterfaceId: eni.NetworkInterfaceId,
+		Groups:             aws.StringSlice(desiredGroups),
+	})
+	return err
+}
+
+//findENIsSupportingTargets find the ID of ENIs that are used to supporting ingress traffic to targets
```

Comment:
`s///findENIsSupportingTargets/// findENIsSupportingTargets/`?

Reply:
If there isn't any, we probably should set up golint to automate the checking of similar styling issue. Also this could be an issue tracked separately.

Reply:
fixed :D actually i have glint in my ide(vscode), not sure why it doesn't complain about this case